tests/test_process_with_context.py

```diff
@@ -168,6 +184,39 @@ def test_process_index(tmpdir):
             ),
             marks=pytest.mark.xfail(raises=ValueError)
         ),
+        pytest.param(  # process_func returns None
+            signal_no_value,
+            signal_with_context_no_value,
+            np.random.random(5 * 44100),
+            44100,
+            audinterface.utils.signal_index(
+                pd.timedelta_range('0s', '3s', 3),
+                pd.timedelta_range('1s', '4s', 3),
+            ),
+            marks=pytest.mark.xfail(raises=RuntimeError),
+        ),
+        pytest.param(  # process_func returns single value
+            signal_single_value,
+            signal_with_context_single_value,
+            np.random.random(5 * 44100),
+            44100,
+            audinterface.utils.signal_index(
+                pd.timedelta_range('0s', '3s', 3),
+                pd.timedelta_range('1s', '4s', 3),
+            ),
+            marks=pytest.mark.xfail(raises=RuntimeError),
+        ),
+        pytest.param(  # process_func returns array with wrong length
+            signal_single_value,
+            lambda sig, fs, starts, ends: [0, 1],
```

Comment:
would it make sense to use a lambda in all cases to avoid introducing functions?

Reply:
Switched to using `lambda` in the new tests